src/node.rs

```diff
@@ -346,6 +236,85 @@ impl<K: ArtKey, V: Default, const MAX_PARTIAL_LEN: usize> ArtNode<K, V, MAX_PART
         }
     }
 
+    #[inline]
+    fn compression(&mut self, prefix_mismatch_pos: usize, key: K, depth: usize, val: V) {
+        let mut old_node = std::mem::replace(self, ArtNode::node4(Box::default()));
```

Comment:
Replace `std::mem::take` with 'std::mem::replace'. When the replace succeeds, **self** becomes **new_node4** and **old_node** is the previous **self**